lib/features/thread/presentation/thread_controller.dart

```diff
@@ -521,11 +524,45 @@ class ThreadController extends BaseController {
     mailboxDashBoardController.dispatchRoute(DashboardRoutes.emailDetailed);
   }
 
-  void selectEmail(BuildContext context, PresentationEmail presentationEmailSelected) {
-    emailList.value = emailList
-      .map((email) => email.id == presentationEmailSelected.id ? email.toggleSelect() : email)
-      .toList();
+  void _shiftSelectEmail(PresentationEmail presentationEmailSelected) {
+    int startIndexAction = 0;
+    int endIndexAction = 0;
+    final indexOfPresentationEmailSelected = emailList.indexWhere((e) => e.id == presentationEmailSelected.id);
+    final indexOfLatestEmailAction = emailList.indexWhere((e) => e.id == latestEmailAction.value?.id);
```

Comment:
what is the value of `Action` in here? please think about the variable, more and more and more